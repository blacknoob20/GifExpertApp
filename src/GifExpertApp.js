import React, { Fragment, useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

const GifExpertApp = ({categoriasDef = []}) => {
    // const categorias = ['One Puch','Samurai X','Dragon Ball'];
    // const [categorias, setCategorias] = useState(['One Puch','Samurai X','Dragon Ball'])
    const [categorias, setCategorias] = useState(categoriasDef);
    
    // const handelAdd = () => {
    //     setCategorias(['Hunter x Hunter',...categorias]);
    //     // setCategorias('Hunter x Hunter'); // Cambia el estado de array a string
    //     // categorias.push('Hunter x Hunter'); // Mala practica
    // }

    return (
        <Fragment>
            <h2>GifExpertApp</h2>
            <AddCategory setCategorias={setCategorias} />
            <hr />
            {/* <button onClick={handelAdd}>Agregar</button> */}
            <ol>
            {
                categorias.map(
                    (categoria) => (
                        <GifGrid
                            key={categoria}
                            categoria={categoria}
                        />
                    )
                )
            }
            </ol>
        </Fragment>
    );
}

export default GifExpertApp;