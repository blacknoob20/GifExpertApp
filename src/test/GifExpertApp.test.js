import React from 'react';
import {shallow} from 'enzyme';
import GifExpertApp from '../GifExpertApp';

describe('Pruebas GifExpertApp', () => {
    test('Debe mostrarse correctamente.', () => {
        const wrapper = shallow(<GifExpertApp />);

        expect(wrapper).toMatchSnapshot();
    });
    
    test('Debe mostrar dos categorias', () => {
        const categorias = ['hunter x hunter', 'full metal'];
        const wrapper = shallow(<GifExpertApp categoriasDef={categorias} />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categorias.length);
    })
    
});
