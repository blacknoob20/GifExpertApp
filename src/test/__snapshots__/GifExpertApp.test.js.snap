// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas GifExpertApp Debe mostrar dos categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <Component
    setCategorias={[Function]}
  />
  <hr />
  <ol>
    <Component
      categoria="hunter x hunter"
      key="hunter x hunter"
    />
    <Component
      categoria="full metal"
      key="full metal"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas GifExpertApp Debe mostrarse correctamente. 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <Component
    setCategorias={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
